java-checks/src/main/java/org/sonar/java/checks/ToStringUsingBoxingCheck.java

```diff
@@ -85,13 +88,41 @@ public void visitNode(Tree tree) {
       return;
     }
     ExpressionTree memberSelectExpression = ((MemberSelectExpressionTree) methodSelect).expression();
+    Optional<ExpressionTree> argumentOfPrimitiveWrapper = getArgumentOfPrimitiveWrapper(memberSelectExpression);
+    if (argumentOfPrimitiveWrapper.isPresent()) {
+      ExpressionTree argument = argumentOfPrimitiveWrapper.get();
+      Type memberSelectType = memberSelectExpression.symbolType();
+      if (isCompatiblePrimitiveType(memberSelectType, argument.symbolType())) {
+        reportIfCompareToOrToString(mit, memberSelectExpression, memberSelectExpression.symbolType().toString(), argument);
+      }
+    }
+  }
+
+  private static boolean isCompatiblePrimitiveType(Type memberSelectType, Type argumentType) {
+    Type memberSelectAsPrimitive = JUtils.primitiveType(memberSelectType);
+    if (memberSelectAsPrimitive != null) {
+      if (argumentType.equals(memberSelectAsPrimitive)) {
+        return true;
+      }
+      if (argumentType.is("int")) {
+        return memberSelectAsPrimitive.is("double")
+          || memberSelectAsPrimitive.is("float")
+          || memberSelectAsPrimitive.is("long");
+      }
+      if (argumentType.is("float")) {
+        return memberSelectAsPrimitive.is("double");
+      }
```

Comment:
I think we miss the `short` type:
```
    byte myByte = 4;
    new Short(myByte).toString(); // Noncompliant {{Call the static method Short.toString(...) instead of instantiating a temporary object.}}
```
Perhaps we can just test `argumentType.isPrimitive()`...

Reply:
Good idea, let's do this!